eth/tracers/api_blocktrace.go

```diff
@@ -91,6 +92,10 @@ func (api *API) createTraceEnvAndGetBlockTrace(ctx context.Context, config *Trac
 			},
 		}
 	} else if config.Tracer != nil {
+		if *config.Tracer == "legacy" {
+			legacyTrace = true
+		}
```

Comment:
no i remember the 'tracer' has some predefined choices.  https://geth.ethereum.org/docs/developers/evm-tracing/built-in-tracers so we should not use "Tracer" to do a mode selection.   We add a field beside EnableMemory? StorageProofFormat: legacy/flatten/union?